src/main/java/org/openpnp/gui/GcodeDriverConsole.java

```diff
@@ -0,0 +1,200 @@
+package org.openpnp.machine.reference.driver.wizards;
+
+import java.awt.FileDialog;
+import java.awt.Toolkit;
+import java.awt.datatransfer.Clipboard;
+import java.awt.datatransfer.DataFlavor;
+import java.awt.datatransfer.StringSelection;
+import java.awt.event.ActionEvent;
+import java.awt.event.KeyAdapter;
+import java.awt.event.KeyEvent;
+import java.awt.Font;
+
+import javax.swing.AbstractAction;
+import javax.swing.Action;
+import javax.swing.JButton;
+import javax.swing.JComboBox;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTextArea;
+import javax.swing.border.TitledBorder;
+import javax.swing.event.DocumentEvent;
+import javax.swing.event.DocumentListener;
+
+import org.openpnp.gui.MainFrame;
+import org.openpnp.gui.components.ComponentDecorators;
+import org.openpnp.gui.support.AbstractConfigurationWizard;
+import org.openpnp.gui.support.Icons;
+import org.openpnp.gui.support.IntegerConverter;
+import org.openpnp.gui.support.DoubleConverter;
+import org.openpnp.gui.support.MessageBoxes;
+import org.openpnp.machine.reference.driver.GcodeDriver;
+import org.openpnp.machine.reference.driver.GcodeDriver.Command;
+import org.openpnp.machine.reference.driver.GcodeDriver.CommandType;
+import org.openpnp.model.Configuration;
+import org.openpnp.model.LengthUnit;
+
+import com.jgoodies.forms.layout.ColumnSpec;
+import com.jgoodies.forms.layout.FormLayout;
+import com.jgoodies.forms.layout.FormSpecs;
+import com.jgoodies.forms.layout.RowSpec;
+import javax.swing.border.EtchedBorder;
+import java.awt.Color;
+import javax.swing.JTextField;
+import org.pmw.tinylog.Logger;
+
+public class GcodeDriverConsole extends AbstractConfigurationWizard {
+    private final GcodeDriver driver;
+
+    public GcodeDriverConsole(GcodeDriver driver) {
+        this.driver = driver;
+        
+        historyLen = 0;
```

Comment:
In the future I'd suggest using a `List<String> (List<String> history = new ArrayList<>())` rather than an array. It removes the history limit, makes it easier to shift elements around and makes it less likely to run into a programming error. No need to change it here and now, just something to consider for the future.